external/build/youtube-playlist.js

```diff
@@ -0,0 +1,204 @@
+      })
+    );
+
+    await fs.writeFile(targetFile, JSON.stringify(result));
+  }
+}
+
+/**
+ * Fetches all the meta data for the playlist and then calls the function
+ * needed for fetching the single videos data
+ * @param {string} id The YouTube channel id
+ * @return {promise} A promise that resolves in the YouTube playlist data
+ * neeeded to display the playlist component
+ */
+
+async function getPlaylistData(id) {
+  const data = [];
+  const response = await youtube?.playlists?.list({
+    part: PART,
+    channelId: id,
+    auth: API_KEY,
+    maxResults: MAX_RESULTS,
+  });
+
+  const playlists = response?.data?.items;
+
+  if (!playlists || playlists.length === 0) {
+    console.warn('No playlist found');
+  } else {
+    await Promise.all(
+      playlists.map(async playlist => {
+        if (playlist.id) {
+          try {
+            const videoRes = await getPlaylistItemData(playlist?.id);
+            data.push({
+              id: playlist.id,
+              title: playlist?.snippet?.title,
+              description: playlist?.snippet?.description,
+              thumbnail: playlist?.snippet?.thumbnails?.medium?.url,
+              updated: playlist?.snippet?.publishedAt,
+              channel: playlist?.snippet?.channelId,
+              videos: videoRes,
+            });
+          } catch (error) {
+            console.error(error);
+            throw new Error('Error fetching the playlist data');
+          }
+        }
+      })
+    );
+  }
+
+  return data;
+}
+
+/**
+ * Fetches all the video meta data the component will need to render correctly the playlist.
+ * @param {string} id The YouTube playlist id
+ * @return {promise} A promise that resolves in the YouTube videos data
+ * needed to display the playlist component
+ */
+
+async function getPlaylistItemData(id) {
+  const data = [];
+  const response = await youtube.playlistItems.list({
+    part: PART,
+    playlistId: id,
+    auth: API_KEY,
+    maxResults: MAX_RESULTS,
+  });
+
+  const videos = response.data.items;
+
+  if (!videos || videos.length === 0) {
+    throw new Error('No playlist videos found');
+  } else {
+    for (const video of videos) {
+      data.push({
+        id: video.id,
+        title: video?.snippet?.title,
+        description: video?.snippet?.description,
+        thumbnail: video?.snippet?.thumbnails?.medium?.url,
+      });
+    }
+  }
+
+  return data;
+}
+
+/**
+ * Fetches all the channel meta data the component will need to render correctly the playlist.
+ * @param {array} id The YouTube channel id
+ * @return {promise} A promise that resolves in the YouTube Channel data needed
+ * to display the playlist component
+ */
+
+async function getChannelData(id) {
+  const data = {};
+  const response = await youtube.channels.list({
+    part: PART,
+    id: id,
+    auth: API_KEY,
+  });
+
+  const channels = response.data.items;
+
+  if (!channels || channels.length === 0) {
+    throw new Error('No channel found');
```

Comment:
Is that an error really? I think I would simply return early, that could also save a try/catch further up 🙂